pkg/api/postage.go

```diff
@@ -164,15 +165,23 @@ type bucketData struct {
 func (s *Service) postageGetStampsHandler(w http.ResponseWriter, r *http.Request) {
 	isAll := strings.ToLower(r.URL.Query().Get("all")) == "true"
 	resp := postageStampsResponse{}
+
 	resp.Stamps = make([]postageStampResponse, 0, len(s.post.StampIssuers()))
 	for _, v := range s.post.StampIssuers() {
+		var isExpired bool
 		exists, err := s.batchStore.Exists(v.ID())
 		if err != nil {
 			s.logger.Debugf("get stamp issuer: check batch: %v", err)
 			s.logger.Error("get stamp issuer: check batch")
 			jsonhttp.InternalServerError(w, "unable to check batch")
 			return
 		}
+		if !exists {
+			state := s.batchStore.GetChainState()
+			if v.Amount().Cmp(state.TotalAmount) <= 0 {
```

Comment:
`Amount` is the amount per chunk and cannot be compared with `TotalAmount` which is the last incremented value of the contract. If the batch does not exist in the batchstore, then it's expired OR we have not received the event of a newly created batch from the chain yet.